src/main/java/org/broadinstitute/hellbender/engine/FeatureManager.java

```diff
@@ -208,6 +211,26 @@ public boolean isEmpty() {
         return featureSources.isEmpty();
     }
 
+
+    /**
+     * This method finds and returns all of the variant headers for each of the features.
+     *
+     * @return A list of all variant headers for features.
+     */
+    public List<VCFHeader> getAllVariantHeaders() {
+        List<VCFHeader> headers = new ArrayList<>();
+        for (Map.Entry<FeatureInput<? extends Feature>, FeatureDataSource<? extends Feature>> entry : featureSources.entrySet()) {
+            final Object header = entry.getValue().getHeader();
+            if ( header != null && header instanceof VCFHeader ) {
+                headers.add((VCFHeader)header);
+            }
+            if (entry.getKey().getFeatureType() == VariantContext.class) {
+                headers.add((VCFHeader)entry.getValue().getHeader());
+            }
```

Comment:
I think this second `if` statement should be deleted -- otherwise you will end up adding each `VCFHeader` twice, won't you?